File: plugins_backup/qr_excel_lookup/adapter.py
# QR / Barcode & Inventory Excel Lookup Adapter (Python)
# Author: M.A.R.K.E.T. AI Systems
import re

def process(input_data):
    msg = input_data.get("user_message", "")
    
    # Mock database
    inventory = {
        "6291038475012": {"item": "شاحن سريع Type-C 65W", "rack": "A-12", "qty": 45, "status": "متاح"},
        "6291038475099": {"item": "سماعة بلوتوث Pro Max", "rack": "B-04", "qty": 3, "status": "مخزون حرج"},
        "QR-9904": {"item": "شاشة حماية ايفون 15 Pro", "rack": "C-01", "qty": 120, "status": "متاح"}
    }
    
    found_code = None
    for code in inventory:
        if code.lower() in msg.lower():
            found_code = code
            break
            
    if found_code:
        item = inventory[found_code]
        return {
            "status": "success",
            "plugin": "qr_excel_lookup",
            "injected_context": f"[نتائج البحث بالباركود {found_code}]:\n- الصنف: {item['item']}\n- الرف/المكان: {item['rack']}\n- الكمية بالمخزن: {item['qty']} قطعة\n- الحالة: {item['status']}",
            "system_instruction_override": "أنت موظف إدارة المخازن والجرد. أبلغ المستخدم بموقع الصنف والكمية المتاحة بدقة."
        }
        
    return {
        "status": "neutral",
        "plugin": "qr_excel_lookup",
        "injected_context": "[الجرد]: لم يتم مسح رمز باركود أو QR صحيح في الرسالة.",
        "system_instruction_override": "اطلب من المستخدم قراءة الباركود أو إدخال الرقم التسلسلي للصنف."
    }

File: plugins_backup/fashion_store/adapter.lua
-- Fashion Store Context & Discount Injector (Lua)
-- Author: M.A.R.K.E.T. AI Systems

function process(input)
    local msg = string.lower(input.user_message or "")
    local customer = (input.metadata and input.metadata.customer_name) or "العميل"
    
    local catalog = {
        ["ts-102"] = { name = "تيشيرت أوفرسايز قطن 100% (أسود/أبيض)", price = 350, discount_price = 295, sizes = "M, L, XL", stock = 14 },
        ["hd-505"] = { name = "سويت شيرت هودي شتوي مبطن", price = 650, discount_price = 520, sizes = "L, XL, XXL", stock = 4 },
        ["pn-204"] = { name = "بنطلون جينز كارجو زيتي", price = 480, discount_price = 420, sizes = "32, 34, 36", stock = 0 }
    }
    
    local matched_code = nil
    for code, item in pairs(catalog) do
        if string.find(msg, code, 1, true) or string.find(msg, string.upper(code), 1, true) then
            matched_code = code
            break
        end
    end
    
    if matched_code then
        local item = catalog[matched_code]
        local stock_status = item.stock > 0 and ("متاح في المخزن (" .. item.stock .. " قطعة)") or "نفذت الكمية حالياً"
        
        return {
            status = "success",
            plugin = "fashion_store",
            injected_context = string.format([[
[بيانات المنتج الدقيقة من السيستم الداخلي]:
- المنتج: %s
- الكود: %s
- السعر الأصلي: %d ج.م
- السعر بالعرض الحالي: %d ج.م (توفير %d ج.م!)
- المقاسات: %s
- المخزون: %s
- العميل: %s
]], item.name, string.upper(matched_code), item.price, item.discount_price, (item.price - item.discount_price), item.sizes, stock_status, customer),
            system_instruction_override = "أنت مساعد خدمة عملاء متجر ملابس أنيق ولبق. أجب على العميل مستخدماً بيانات المنتج أعلاه فقط بدقة تامة وبدون أي تأليف."
        }
    end
    
    if string.find(msg, "شحن", 1, true) or string.find(msg, "توصيل", 1, true) then
        return {
            status = "success",
            plugin = "fashion_store",
            injected_context = "[سياسة الشحن]: التوصيل لجميع المحافظات خلال 2-4 أيام عمل بـ 45 ج.م. الشحن مجاني للطلبات فوق 600 ج.م.",
            system_instruction_override = "أخبر العميل بسياسة التوصيل بأسلوب ترحيبي مشجع."
        }
    end
    
    return {
        status = "neutral",
        plugin = "fashion_store",
        injected_context = "[المتجر]: لم يذكر العميل كود منتج محدد. اسأله بلطف عن القطعة أو الكود لمساعدته.",
        system_instruction_override = "رحب بالعميل واطلب منه تحديد القطعة أو كود المنتج مثل TS-102."
    }
end
